NLSY/nlsy_lib.R

```diff
@@ -243,6 +243,95 @@ nlsy_get_highest_grade_completed_df = function()
     return(data)
 }
 
+#' Returns marriage status the last month of a given year
+nlsy_get_marstat_df = function()
+{
+    data = nlsydf |>
+        select(
+            id =        "PUBID_1997",
+            starts_with('MAR_STATUS_')
+        ) |>
+        pivot_longer(
+            starts_with('MAR_STATUS_'),
+            names_to=c('year', 'month'),
+            names_pattern='MAR_STATUS_(\\d\\d\\d\\d)\\.(\\d\\d)_XRND',
+            values_to='mar_status'
+        ) |>
+        mutate(
+            year=as.integer(year),
+            month=as.integer(month)
+        ) |>
+        filter(month == 12) |>
+        group_by(id) |>
+        complete(year=1997:2021) |>
```

Comment:
Let's keep 2019 as the last year. I think that 2021 was added after we started the project and I still haven't had a chance to look at this addition.

Reply:
Fixed!